### PR TITLE
Fix crash when filtering on implicit `pk` field on abstract model

### LINKED ISSUE #1189: Plugin crash on abstract model filtering with custom Manager
python 3.10
django 4.1
mypy 0.982
django-stubs HEAD (a78c592)
django-stubs-ext HEAD (a78c592)

Crash in plugin when using an abstract model with a custom manager:

```
./myproj/app/models.py:15: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.982
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 4619, in accept
  File "mypy/nodes.py", line 1822, in accept
  File "mypy/checkexpr.py", line 385, in visit_call_expr
  File "mypy/checkexpr.py", line 505, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 1155, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 1238, in check_call
  File "mypy/checkexpr.py", line 1438, in check_callable_call
  File "mypy/checkexpr.py", line 995, in apply_function_plugin
  File "django-stubs/mypy_django_plugin/transformers/orm_lookups.py", line 40, in typecheck_queryset_filter
    lookup_type = django_context.resolve_lookup_expected_type(ctx, model_cls, lookup_kwarg)
  File "django-stubs/mypy_django_plugin/django/context.py", line 388, in resolve_lookup_expected_type
    lookup_parts, field_parts, is_expression = query.solve_lookup_type(lookup)
  File "/code/myproj/.venv/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1217, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
  File "/code/myproj/.venv/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1656, in names_to_path
    name = opts.pk.name
AttributeError: 'NoneType' object has no attribute 'name'
./myproj/app/models.py:15: : note: use --pdb to drop into pdb
```

Minimal reproduction example (with or without django_stubs_ext.monkeypatch):

NB: This needs to be in a models file in an installed app.
```
from django.db import models


class BaseManager(models.Manager):
    pass


class BaseModel(models.Model):
    objects = BaseManager()

    class Meta:
        abstract = True

    def lock(self) -> None:
        type(self).objects.filter(pk=self.pk).select_for_update().values_list("pk")


class MyModel(BaseModel):
    pass

```

### PR DESCRIPTION
**Pull request by flaeppe**
# I have made things!

Avoid crashing when queryset filters on an implicit `pk` field for a model, when model is abstract.

I tried to figure out #1261 and while type checking Django's test suite I came across the same error as in #1189. But reproduced when filtering on `django.contrib.auth.models.AbstractUser`.

## Related issues

Closes #1189 

I wasn't able to reproduce with the example found in the linked issue though.

### CODE REVIEW

File: mypy_django_plugin/django/context.py
Comment (intgr) [inline mypy_django_plugin/django/context.py]: `lookup.startswith("pk")` can match other column names, e.g. someone may abbreviate `private_key` as `pkey` which starts with `pk`

Does `solve_lookup_type()` fail in this situation? If not, the `lookup_parts` variable seems like better way to detect this.

Comment (flaeppe) [reply mypy_django_plugin/django/context.py]: Yeah, true. While quite unlikely to also match on `.pk is None` we should still avoid it.

`solve_lookup_type()` is what's crashing. We could probably change the statement to:

```python
if lookup == "pk" or lookup.startswith("pk__") and query.get_meta().pk is None:
    ...
```

Comment (flaeppe) [reply mypy_django_plugin/django/context.py]: While there's also the option to mimic Django

```python
lookup_splitted, _ = lookup.split(LOOKUP_SEP, 1)
if lookup_splitted == "pk" and query.get_meta().pk is None:
    ...
```

Comment (intgr) [reply mypy_django_plugin/django/context.py]: Either of these look fine to me. I'll leave it up to you.

Comment (flaeppe) [reply mypy_django_plugin/django/context.py]: I went with the first comment to avoid an import of`LOOKUP_SEP`

**Review by intgr — APPROVED** [review]
Looks good to me, waiting for CI.

**Comment by simonpercivall** [comment]
Actually, #1189 wasn't closed by this. My example was faulty, it was a `.values_list("pk")` that causes the issue.

**Pull request merged**
